annotation/app/tasks/services.py

```diff
@@ -487,3 +491,59 @@ def create_export_csv(
 
     today = datetime.today().strftime("%Y%m%d")
     return f"annotator_stats_export_{today}.csv", binary
+
+
+class _MetricScoreTuple(NamedTuple):
+    task_from: int
+    task_to: int
+    score: float
+
+
+def compare_agreement_scores(
+    agreement_score_response: List[AgreementScoreServiceResponse],
+    min_match: float,
+) -> AgreementScoreComparingResult:
+    # firstly get unique pairs of task agreement score metrics, example:
+    # (1, 2, 0.95), (2, 1, 0.95) where 1 and 2 are task ids
+    # there is no need to store both of them in db since they are the same
+    # so transform (1, 2, 0.95), (2, 1, 0.95) to (1, 2, 0.95)
+    unique_scores: Set[_MetricScoreTuple] = set()
+    for entity in agreement_score_response:
+        task_from_id: int = entity.task_id
+        scores: List[ResponseScore] = pydantic.parse_obj_as(
+            List[ResponseScore],
+            entity.agreement_score.get("scores", [])
+            if entity.agreement_score
+            else [],
+        )
+        for score in scores:
+            el = _MetricScoreTuple(
+                task_from_id, score.task_id, score.agreement_score
+            )
+            unique_scores.add(
+                _MetricScoreTuple(
+                    min(el.task_from, el.task_to),
+                    max(el.task_from, el.task_to),
+                    el.score,
+                )
+            )
```

Comment:
May you move it to another function, like `get_unique_scores(scores: List[ResponseScore], task_id: int) -> Set[_MetricScoreTuple]` to make this function a bit less "god-like" and increase readability. Optional

Reply:
Moving this block of code in another function is pointless in my opinion because we ain't gonna reuse it anywhere else, so we'll just create a redundant function.
I believe this code should be encapsulated into this function.